src/v/cloud_storage/types.cc

```diff
@@ -246,6 +248,9 @@ void scrub_segment_meta(
     }
 }
 
+// Limit on number of anomalies that can be stored in the manifest
+static constexpr size_t max_number_of_manifest_anomalies = 100;
```

Comment:
How do you feel about making this configurable?

Reply:
I think it's ok for this value to be hardcoded. I can't imagine a situation when we need to tweak it manually. The actual value doesn't really matter because in most practical situations this value should be low (ideally zero).